File: scolioscan-pro-fastapi/backend/app/schemas/alarm.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from uuid import UUID


class AlarmCreate(BaseModel):
    user_uuid: UUID
    alarm_type: int
    title: str
    content: str


class AlarmResponse(BaseModel):
    id: int
    user_uuid: UUID
    alarm_type: int
    title: str
    content: str
    read_at: Optional[datetime]
    created_at: datetime

    class Config:
        from_attributes = True

File: scolioscan-expo/src/types/alarm.ts
export interface AlarmResponse {
  id: number;
  user_uuid: string;
  alarm_type: number;
  title: string;
  content: string;
  read_at: string | null;
  created_at: string;
}

export interface AlarmUnreadCountResponse {
  count: number;
}
